server/src/main/java/org/elasticsearch/index/query/NestedQueryBuilder.java

```diff
@@ -260,6 +260,34 @@ protected int doHashCode() {
 
     @Override
     protected Query doToQuery(SearchExecutionContext context) throws IOException {
+        Query innerQuery;
+        NestedObjectMapper mapper = context.nestedLookup().getNestedMappers().get(path);
```

Comment:
I think this change breaks a nested query test. My best guess is because this lookup is repeated in `toQuery` (see line 297)

Reply:
I ripped out the internals - I have corrected it now.